diff -Naur upstream/stb_image.h zxing/stb_image.h
--- upstream/stb_image.h	2021-12-08 18:22:56.724466161 -0500
+++ zxing/stb_image.h	2021-12-08 18:23:15.084657043 -0500
@@ -1725,7 +1725,11 @@
 
 static stbi_uc stbi__compute_y(int r, int g, int b)
 {
+#if 0 // ori
    return (stbi_uc) (((r*77) + (g*150) +  (29*b)) >> 8);
+#else // zxing (see ReadBarcode.cpp:RGBToGray)
+   return (stbi_uc) ((306 * r + 601 * g + 117 * b + 0x200) >> 10);
+#endif
 }
 #endif
 
